state/factory/statedb.go

```diff
@@ -51,11 +50,11 @@ type stateDB struct {
 }
 
 // StateDBOption sets stateDB construction parameter
-type StateDBOption func(*stateDB, config.Config) error
+type StateDBOption func(*stateDB, *Config) error
```

Comment:
i think let's use `Config`,  to be consistent with `factory.go` and other components